check/check.go

```diff
@@ -77,37 +92,140 @@ type Group struct {
 
 // Run executes the audit commands specified in a check and outputs
 // the results.
-func (c *Check) Run() {
+func (c *Check) Run(definedConstraints map[string][]string) {
+	var subCheck *BaseCheck
+	if c.SubChecks == nil {
+		subCheck = &BaseCheck{
+			Commands:    c.Commands,
+			Tests:       c.Tests,
+			Type:        c.Type,
+			Audit:       c.Audit,
+			Remediation: c.Remediation,
+		}
+	} else {
+		subCheck = getFirstValidSubCheck(c.SubChecks, definedConstraints)
 
-	if c.Type == "skip" {
-		c.State = INFO
+		if subCheck == nil {
+			c.State = WARN
+			glog.V(1).Info("Failed to find a valid sub check, check ", c.ID)
+			return
+		}
+	}
+
+	var out bytes.Buffer
```

Comment:
See comment below about making `out` and `errmsgs` return values rather than parameters. Then you could also just have `:=`  to set them from `runAuditCommands()` rather than having to explicitly declare them as variables.

Reply:
Can't do it because of the c.State param